src/plots/cartesian/layout_attributes.js

```diff
@@ -465,6 +465,36 @@ module.exports = {
             'Only has an effect if `anchor` is set to *free*.'
         ].join(' ')
     },
+    categorymode: {
+        valType: 'enumerated',
+        values: [
+            'trace', 'category ascending', 'category descending', 'array'
```

Comment:
Maybe we should include truncated versions of those values as well e.g. `'cat asc'`, `'cat des'` ?